src/polly/src/epoll.rs

```diff
@@ -63,48 +63,71 @@ bitflags! {
     }
 }
 
+impl EventSet {
+    /// Checks if `self` contains only the bits in `other_set`.
+    pub fn contains_only(self, other_set: EventSet) -> bool {
+        (self - other_set).is_empty()
+    }
+}
+
 /// Wrapper over 'libc::epoll_event'.
 ///
 // We are using `transparent` here to be super sure that this struct and its fields
 // have the same alignment as those from the `epoll_event` struct from C.
 #[repr(transparent)]
 #[derive(Clone, Copy)]
-pub struct Event(epoll_event);
+pub struct EpollEvent(epoll_event);
 
-impl Deref for Event {
+impl Deref for EpollEvent {
     type Target = epoll_event;
     fn deref(&self) -> &Self::Target {
         &self.0
     }
 }
 
-impl Default for Event {
+impl Default for EpollEvent {
     fn default() -> Self {
-        Event(epoll_event {
+        EpollEvent(epoll_event {
             events: 0u32,
             u64: 0u64,
         })
     }
 }
 
-impl Event {
+impl EpollEvent {
     /// Create a new epoll_event instance with the following fields: `events`, which contains
     /// an event mask and `data` which represents a user data variable. `data` field can be
     /// a fd on which we want to monitor the events specified by `events`.
-    pub fn new(events: EventType, data: u64) -> Self {
-        Event(epoll_event {
+    pub fn new(events: EventSet, data: u64) -> Self {
+        EpollEvent(epoll_event {
             events: events.bits(),
             u64: data,
         })
     }
 
+    /// Returns the `events` from `libc::epoll_event`.
     pub fn events(&self) -> u32 {
         self.events
     }
 
+    /// Returns the `EventSet` corresponding to `epoll_event.events` or
+    /// None if the events are invalid.
+    pub fn event_set(&self) -> Option<EventSet> {
```

Comment:
Hmm, it might make sense actually. From my understanding, the event_set is invalid is something is wrong with the kernel or the user specified a wrong value there. Since the inner `epoll_event` cannot be changed from outside of the implementation of EpollEvent, we can do that unwrap inside this function and just add a `# Panics` section to the documentation. Does that sound sane?